src/main/scala/calc/Compiler.scala

```diff
@@ -66,7 +70,7 @@ object Compiler {
    *
    *  This is the main method you have to implement.
    */
-  def compileExpr(tree: Tree): irt.Tree = {
+  def compileExpr(tree: Tree, listId: List[Ident] = List()): irt.Tree = {
```

Comment:
I would a recommend a `Set` of `String` (`Ident`s are not meant to be compared), but otherwise, yes, that's the idea.